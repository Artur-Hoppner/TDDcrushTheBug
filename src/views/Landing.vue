<template>
  <div></div>
</template>

<script>
export default {
  name: 'Landing',
  components: {}
};

</script>

<style></style>
