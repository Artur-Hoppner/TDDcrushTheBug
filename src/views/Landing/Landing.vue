<template>
  <div class="landing-container">
    <div class="introduction-container">
      <img class="garden-image" src="@/assets/garden.jpg" alt="Garden Image" />
      <div class="paragraph-container">
        <p class="paragraph-introduction">
          Repel, Remove or Blast your bugs!
          <br />We have the solution for all of your pest problems. With our
          products you will be able to get the same results as if you hired a
          professional exterminator.
          <a
            class="link-to-shop"
            id="link-to-shop"
            href
            @click="goTo('/shopping')"
          >Check out our products</a>, and become a professional exterminator yourself
          <br />
          <br />We are a small family business that have a long tradition of
          hating insects!
        </p>
      </div>
    </div>

    <div class="exterminator-container">
      <img
        class="exterminator"
        src="@/assets/backgroundImages/exterminator-5.png"
        alt="exterminator"
      />
      <img class="ant2" src="@/assets/backgroundImages/ant-1.png" alt="ant 2" />
    </div>
  </div>
</template>

<script>
import { mapActions } from 'vuex';
export default {
  name: 'Landing',
  methods: {
    ...mapActions(['goTo'])
  }
};
</script>

<style lang="scss" scoped>
.landing-container {
  margin-top: 5px;
  display: flex;
  flex-direction: column;
  align-content: center;
  .introduction-container {
    margin: 5px 0 0 20px;
    display: flex;
    flex-direction: row;
    @media (max-width: 1000px) {
      flex-direction: column-reverse;
    }
    .paragraph-container {
      display: flex;
      align-self: center;
      padding: 1vw;
      border: #a5d1bc8e solid 1px;
      margin: 30px 5vw 2vh 5vw;
      .paragraph-introduction {
        font-family: 'Raleway', sans-serif;
        font-size: 20px;
        line-height: 1.6;
        .link-to-shop {
          text-decoration: none;
          color: #a5d1bc;
          font-weight: 900;
        }
      }
    }
    .garden-image {
      // opacity: 85%;
      height: 25vw;
      width: 40vw;
      border-radius: 1%;
      @media (max-width: 1000px) {
        align-self: center;
        height: 40vw;
        width: 60vw;
        margin-bottom: 1vh;
      }
    }
  }
}
.exterminator-container {
  display: flex;
  justify-content: flex-end;
  align-items: flex-end;
  .ant2 {
    width: 10vw;
    height: 6vw;
  }
  .exterminator {
    width: 15vw;
    height: 15vw;
  }
}
</style>
