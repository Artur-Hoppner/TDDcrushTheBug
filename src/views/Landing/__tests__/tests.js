import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Landing from '@/views/Landing/Landing.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Test Landing', () => {
  test('Test if /Landing is rendered', () => {
    //Arrange
    const wrapper = shallowMount(Landing, { localVue });
    //Assert
    expect(wrapper.element).toMatchSnapshot();
  });

  test('test if link action goTo is activated on click', async () => {
    //Arrange
    const actions = {
      goTo: jest.fn()
    };
    const store = new Vuex.Store({
      actions
    });
    const wrapper = shallowMount(Landing, {
      localVue,
      store
    });
    // Act
    const input = wrapper.find('#link-to-shop');
    console.log(input, 'input data');
    await input.trigger('click');
    await localVue.nextTick();
    // Assert
    expect(actions.goTo).toHaveBeenCalled();
  });
});
