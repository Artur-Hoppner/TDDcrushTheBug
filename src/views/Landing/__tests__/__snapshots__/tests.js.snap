// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Landing Test if /Landing is rendered 1`] = `
<div
  class="landing-container"
>
  <div
    class="introduction-container"
  >
    <img
      alt="ant 1"
      class="garden-image"
      src="@/assets/garden.jpg"
    />
     
    <div
      class="paragraph-container"
    >
      <p
        class="paragraph-introduction"
      >
        
        Repel, Remove or Blast your bugs!
        
        <br />
        We have the solution for all of your pest problems. With our
        products you will be able to get the same results as if you hired a
        professional exterminator.
        
        <a
          class="link-to-shop"
          href=""
          id="link-to-shop"
        >
          Check out our products
        </a>
        , and become a professional exterminator yourself
        
        <br />
         
        <br />
        We are a small family business that have a long tradition of
        hating insects!
      
      </p>
    </div>
  </div>
   
  <div
    class="exterminator-container"
  >
    <img
      alt="exterminator"
      class="exterminator"
      src="@/assets/backgroundImages/exterminator-5.png"
    />
     
    <img
      alt="ant 2"
      class="ant2"
      src="@/assets/backgroundImages/ant-1.png"
    />
  </div>
</div>
`;
